ls1intum/Artemis
# `Grading`: Access and filter results view via score distribution chart

### Checklist
#### General


#### Client

### Motivation and Context


This PR makes the results overview of an exercise accessible via the score distribution in the exercise statistics. Additionally, a filter depending on the selected interval in the chart applies to the shown results.
### Description


### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Exercise of each type (except quiz) and many different results (i.e. students got different scores for their participation)

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to the exercise detail page
4. Access the exercise statistics
5. Select any bar in the score distribution
6. Check that the occurring result overview is filtered accordingly and the corresponding filter setting is displayed

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| exercise-scores.component.ts | 77.92 | 96.92 |
| statistics-score-distribution-graph.component.ts | 87.5 | 92.59 |


### Screenshots


![image](https://user-images.githubusercontent.com/80622272/177129396-99e3b77c-0e6f-4bbe-b159-872468fc8a72.png)
